### PR TITLE
Implement schedule as template function.

### PR DESCRIPTION
**Pull request by runer53248**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: schedule
🏅 Granted 12 XP for @runer53248

**Comment by ziobron** [comment]
⏱ 3 XP extra for @runer53248 for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: schedule-bonus
🏅 Granted 10 XP for @runer53248

**Comment by ziobron** [comment]
⏱ 3 XP extra for @runer53248 for on-time delivery

File: homework/schedule/schedule.hpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <chrono>
+#include <functional>
+#include <thread>
+
+template<typename T, typename ...Args>
+void schedule(T&& fun, std::chrono::duration<double> time, Args&&... args){
+    std::this_thread::sleep_for(time);
+    std::invoke(fun, std::forward<Args>(args)...);
```
Comment (BarTes8) [inline homework/schedule/schedule.hpp:9]: what are the advantages of using here std::invoke and std::forward?

**Comment by ziobron** [comment]
Thanks @BarTes8 for Code Review.
🏅 1 XP granted

File: homework/schedule/schedule.hpp (hunk 1 above)
Comment (runer53248) [reply homework/schedule/schedule.hpp:9]: The std:: invoke allows you to call methods of the class. Example:
    schedule(&Foo::print, 1s, foo, 15);
The std::forward function  allows you to pass temporary values to function without copy or assigment.
The std::forward function acts as a conditional cast to T&& when the deduced template parameter is a non-reference type.
This makes the schedule function more universal to use.

**Comment by ziobron** [comment]
Thanks @BarTes8 for Code Review.
🏅 1 XP granted

File: homework/schedule/schedule.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline homework/schedule/schedule.hpp:9]: Perfect One! This is 100% what I expected from senior software ;p

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**